src/modules/flow/power-supply/power-supply.json

```diff
@@ -0,0 +1,166 @@
+{
+  "$schema": "http://solettaproject.github.io/soletta/schemas/node-type-genspec.schema",
+  "name": "power-supply",
+  "meta": {
+    "author": "Intel Corporation",
+    "license": "BSD-3-Clause",
+    "version": "1"
+  },
+  "types": [
+    {
+      "category": "power-supply",
+      "description": "Send list of all power supplies attached to board.",
+      "in_ports": [
+        {
+          "data_type": "any",
+          "description": "Trigger to get list of power supplies.",
+          "methods": {
+            "process": "get_list_process"
+          },
+          "name": "TRIGGER"
+        },
+        {
+          "data_type": "string",
+          "description": "Name of type to be used to filter power supplies. It can be 'Any', 'Unknown', 'Battery', 'UPS', 'Mains', 'USB', 'USB_DCP', 'USB_CDP', 'USB_ACA'.",
+          "methods": {
+            "process": "set_type_process"
+          },
+          "name": "TYPE"
+        }
+      ],
+      "methods": {
+        "open": "get_list_open"
+      },
+      "name": "power-supply/get-list",
```

Comment:
we should keep a standard name with oic as well, like discover or get-list all around.